blog/2025-02-25-saml-idp-initiated-flow.md

```diff
@@ -0,0 +1,93 @@
+
+### Okta setup example
+
+To facilitate a smoother setup process, we will provide a complete example. The setup will utilize Okta as the identity provider:
+
+**Okta:**
+
+Login in your Okta tenant and configure a new application. 
+
+![Okta Application Start](/blog/saml/idp-init/okta-application-start.png)
+
+You will need to specify two things:
+
+`Single sign-on URL`: http://localhost:8080/realms/test-realm/broker/okta-broker/endpoint/clients/okta-client \
+`Audience URI`: http://localhost:8080/realms/test-realm 
+
+![Okta Application Setup](/blog/saml/idp-init/okta-application-setup.png)
+
+The identity provider `redirect url` differs from what we typically observe in  Identity Provider from the Keycloak console. Based on the documentation for [IDP Initiated Login](https://www.keycloak.org/docs/latest/server_admin/#idp-initiated-login), the path `{brokerRedirectUrl}/clients/okta-client` indicates the `client-id` that is intended to maintain the service-provider application session. This will make more sense in the following steps.
+
+**SAML 2.0 Identity provider:**
+
+The SAML identity provider (e.q: ```okta-broker```) from the example above can be created by importing the metadata.xml from the application you just created in Okta. For this example we need the alias: ```okta-broker```.
+
+![Identity Provider Setup](/blog/saml/idp-init/identity-provider-setup.png)
+
+**Service provider application**
+
+For this we created a simple SpringBoot app which has a `ACS` endpoint: ```/login/saml2/sso``` and a `entityId`: ```/saml2/metadata``` . Source code: [SpringBoot SAML application](https://github.com/p2-inc/examples/saml2/idp-initiated)
+
+**Keycloak realm client**
+
+We need to create new SAML 2.0 client in `test-realm`. This can be done by importing the metadata from the Service Provider application. 
+
+![Generic client](/blog/saml/idp-init/generic-client.png)
+
+Now we need to return to the documentation: [IDP Initiated Login](https://www.keycloak.org/docs/latest/server_admin/#idp-initiated-login). In order to configure the IDP-initiated flow, a special field needs to be specified ```IDP-Initiated SSO URL```. \
+IDP-Initiated SSO URL name: `okta-client` 
+
+The field is part of the `{brokerRedirectUrl}/clients/okta-client` url. As we can see the value is different from that of the `clientId` of the client we just created.
+
+**Testing**
+
+To test this, please visit the OKTA end user dashboard and select the application we created in the first step.
+
+![Okta flow start](/blog/saml/idp-init/okta-init-flow.png)
+
+For debugging purposes, you might consider using the SAML Tracer browser extension. If you take a moment to check the requests from the flow, you'll notice that it contains only SAMLResponse messages. This is a specific characteristic of the IDP-initiated flow.
```

Comment:
Formatting nit: Make sure we are consistent with how we refer to "IDP-initiated" or "IdP-initiated". Pick one and stick to it.